src/core/image.rs

```diff
@@ -1,16 +1,19 @@
 use crate::core::colour_models::*;
 use crate::core::traits::PixelBound;
 use ndarray::prelude::*;
-use ndarray::s;
+use ndarray::{s, Data, DataMut, OwnedRepr, RawDataClone, ViewRepr};
 use num_traits::cast::{FromPrimitive, NumCast};
 use num_traits::Num;
-use std::marker::PhantomData;
+use std::{fmt, hash, marker::PhantomData};
+
+pub type Image<T, C> = ImageBase<OwnedRepr<T>, C>;
+pub type ImageView<'a, T, C> = ImageBase<ViewRepr<&'a T>, C>;
```

Comment:
Shame https://github.com/rust-lang/rfcs/pull/2515 hasn't landed yet - it would have been nicer to phrase these as:
```
pub type Image<T, C> = ImageBase<impl DataMut<T>, C>;
pub type ImageView<'a, T, C> = ImageBase<impl Data<&'a T>, C>;
```

Reply:
Yeah, there are also bits of the code that would be nicer with const generics as well, I've just accepted eventual refactoring for the new things I want to use